### PR TITLE
Add LLMs support for Anyscale Service

### PR DESCRIPTION
**Pull request by kylehh**
# Your PR Title (What it does)

Add Anyscale service integration under LLM

## Before submitting

Added integration test code. and sample code for using Ray to distribute service endpoint queries

## Who can review?

Community members can review the PR once tests pass. Tag maintainers/contributors who might be interested:
 @hwchase17 @agola11

### CODE REVIEW

File: langchain/llms/anyscale.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        return {
+            "anyscale_service_url": self.anyscale_service_url,
+            "anyscale_service_route": self.anyscale_service_route
+            #**{"model_kwargs": _model_kwargs},
+        }
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of llm."""
+        return "Anyscale LLM"
```
Comment (dev2049) [inline langchain/llms/anyscale.py:81]: convention so far as has been to use kebab-case, and probably don't need LLM in there. so really just "anyscale"

File: langchain/llms/anyscale.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+	    prompt_list=[]
+            @ray.remote
+            def send_query(llm, prompt):
+                resp = llm(text)
+                return resp
+            futures = [send_query.remote(anyscalellm, prompt) for promt in prompt_list]
+            results = ray.get(futures)
+    """
+
+    client: Any  #: :meta private:
```
Comment (dev2049) [inline langchain/llms/anyscale.py:33]: is this instantiated / used anywhere?

File: langchain/llms/anyscale.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+class AnyscaleLLM(LLM):
+    """Wrapper around Anyscale Services.
+    To use, you should have the environment variable ``ANYSCALE_SERVICE_URL`` set with your Anyscale Service URL, or pass it as a named parameter to the constructor.
+    
+    Example:
+        .. code-block:: python
+            from langchain.llms import AnyscaleLLM
+            anyscalellm = AnyscaleLLM(anyscale_service_url="SERVICE_URL",
+                                anyscale_service_route="SERVICE_ROUTE",
+                                anyscale_service_token="SERVICE_TOKEM")
```
Comment (dev2049) [inline langchain/llms/anyscale.py:21]: SERVICE_TOKEN

File: langchain/llms/anyscale.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        values['anyscale_service_token'] = anyscale_service_token
+        return values
+
+    @property
+    def _identifying_params(self) -> Mapping[str, Any]:
+        """Get the identifying parameters."""
+        return {
+            "anyscale_service_url": self.anyscale_service_url,
+            "anyscale_service_route": self.anyscale_service_route
+            #**{"model_kwargs": _model_kwargs},
```
Comment (dev2049) [inline langchain/llms/anyscale.py:75]: uncomment or delete

File: langchain/llms/anyscale.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+        """Return type of llm."""
+        return "Anyscale LLM"
+
+    def _call(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+    ) -> str:
+        """Call out to HuggingFace Hub's inference endpoint.
```
Comment (dev2049) [inline langchain/llms/anyscale.py:89]: update docstring

**Review by dev2049 — COMMENTED** [review]
This is awesome! Couple of smaller comments. Could we also add a simple example notebook (see `docs/modules/models/llms/integrations/openai.ipynb` for an example)?  We may also want to add a quick note about anyscale in docs/ecosystem section

File: langchain/llms/anyscale.py (hunk 2 above)
Comment (kylehh) [reply langchain/llms/anyscale.py:33]: No, I will remove it.

File: langchain/llms/anyscale.py (hunk 5 above)
Comment (kylehh) [reply langchain/llms/anyscale.py:89]: done

File: langchain/llms/anyscale.py (hunk 4 above)
Comment (kylehh) [reply langchain/llms/anyscale.py:75]: done

File: langchain/llms/anyscale.py (hunk 3 above)
Comment (kylehh) [reply langchain/llms/anyscale.py:21]: done

File: langchain/llms/anyscale.py (hunk 1 above)
Comment (kylehh) [reply langchain/llms/anyscale.py:81]: Updated. Also modified class name just to be "Anyscale" to follow the convention

**Comment by kylehh** [comment]
Added both example notebook and quick note under docs/ecosystem.
Thanks!

File: docs/ecosystem/anyscale.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
+# Anyscale
+
+This page covers how to use the Anyscale ecosystem within LangChain.
+It is broken into two parts: installation and setup, and then references to specific Anyscale wrappers.
+
+## Installation and Setup
+- Get an Anyscale Service URL, route and API key and set them as environment variables (`ANYSCALE_SERVICE_URL`,`ANYSCALE_SERVICE_ROUTE`, `ANYSCALE_SERVICE_TOKEN`)
```
Comment (dev2049) [inline docs/ecosystem/anyscale.md:7]: Any good links we can provide here? e.g. to a Get Started page for Anyscale or something like that

**Review by dev2049 — APPROVED** [review]
one small comment, otherwise looks good to me!

File: docs/ecosystem/anyscale.md (hunk 6 above)
Comment (kylehh) [reply docs/ecosystem/anyscale.md]: Sure. Just added the docs link and also reformatted anyscale.py with Black

**Comment by kylehh** [comment]
Thanks for the review @dev2049 . Btw. I noticed there is a typo,and  it says "Anysacle" instead of "Anyscale" on this [page](https://python.langchain.com/en/latest/modules/models/llms/integrations/anyscale.html), how do I fix it?

**Comment by dev2049** [comment]
> Thanks for the review @dev2049 . Btw. I noticed there is a typo,and it says "Anysacle" instead of "Anyscale" on this [page](https://python.langchain.com/en/latest/modules/models/llms/integrations/anyscale.html), how do I fix it?

that's just showing the notebook here `docs/modules/models/llms/integrations/anyscale.ipynb`, so the typo must be in there

**Comment by kylehh** [comment]
I see. Im fixing it. Thanks!

**Comment by kylehh** [comment]
A new PR #4526 was created to fix this typo. Thanks!

**Pull request merged**